libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -482,6 +483,7 @@ TickType_t xIdentifier = 0;
 	}
 
 	return ulIPAddress;
+  }
```

Comment:
Maybe I'm reading this wrong, but the return should come after this brace, yes?  Before, when I said it should be a boolean, I meant `( pcHostName != NULL )`, so this code exhibits the same shortcoming.

Reply:
Yes, the brace comes before the return. Sorry for this. `if (pcHostName)` is equivalent to the what you said, but I changed it to the explicit `NULL` check.

Reply:
MISRA rule 14.4: "The controlling expression of an if statement and the controlling expression of an iteration-statement shall have essentially Boolean type."